packages/base-contract/src/index.ts

```diff
@@ -14,9 +14,23 @@ export class BaseContract {
         values: any[],
         transformation: (type: string, value: any) => any,
     ): any {
-        return _.map(values, (value: any, i: number) =>
-            BaseContract._transformTypedData(abis[i].type, value, transformation),
-        );
+        return _.map(values, (value: any, i: number) => {
```

Comment:
I've added a TODO for myself to make this generic and support arbitraty nested tuples.